src/iosched/unified.c

```diff
@@ -1359,17 +1461,28 @@ void _unified_process_data_queue(enum request_state queue, struct unified_data *
  * @return 0 on success or a negative value on error. This function always succeeds
  *         if alloc is false.
  */
-int _unified_get_dentry_priv(struct dentry *d, struct dentry_priv **dentry_priv,
-	struct unified_data *priv)
+int _unified_get_dentry_priv(struct dentry *d,
+							 struct dentry_priv **dentry_priv,
+							 struct unified_data *priv)
 {
 	int ret;
 	size_t max_filesize;
 	struct dentry_priv *dpr;
 
+	acquireread_mrsw(&d->meta_lock);
 	if (d->iosched_priv) {
-		*dentry_priv = d->iosched_priv;
+		dpr = d->iosched_priv;
+		ltfs_mutex_lock(&dpr->ref_lock);
+		dpr->numhandles++;
+		if (dentry_priv) {
+			*dentry_priv = d->iosched_priv;
```

Comment:
You could make it '*dentry_priv = dpr', since that's the alias you were using just a few lines above

Reply:
Fixed in my branch.